### DISCUSSION TITLE
Improve misleading "depName mismatch" trace with incorrect newDepName when currentValue regexp not supporting replacement with newVersion

### DISCUSSION THREAD

**Discussion by gberche-orange** [renovatebot/renovate#22522] [Suggest an Idea]
### Tell us more.

With the regexp manager, when the currentValue regexp is properly matching the current version, but fails to replace the new version, then the debug traces misleadingly reports a depName mismatch.

```
DEBUG: Starting search at index 474 (branch="renovate/cloudfoundry-community-haproxy-boshrelease-12.x")
{
  "packageFile": "remote-r4-depls/root-deployment.yml"
  "depName": "cloudfoundry-community/haproxy-boshrelease"
}

DEBUG: Found match at index 474 (branch="renovate/cloudfoundry-community-haproxy-boshrelease-12.x")
{
  "packageFile": "remote-r4-depls/root-deployment.yml"
  "depName": "cloudfoundry-community/haproxy-boshrelease"
}

DEBUG: depName mismatch (branch="renovate/cloudfoundry-community-haproxy-boshrelease-12.x")
{
  "manager": "regex"
  "packageFile": "remote-r4-depls/root-deployment.yml"
  "currentDepName": "cloudfoundry-community/haproxy-boshrelease"
  "newDepName": "orange-cloudfoundry/networking-release"
}

WARN: Error updating branch: update failure (branch="renovate/cloudfoundry-community-haproxy-boshrelease-12.x")
```

Additionally, the dependency dashboards mentions a `WARN: Error updating branch: update failure`

The preceding trace properly indicates the newVersion that will be attempted to be replaced
```
 {
            "depName": "cloudfoundry-community/haproxy-boshrelease",
            "currentValue": "11.17.4",
            "datasource": "github-tags",
            "versioning": "loose",
            "extractVersion": "^v(?<version>.*)",
            "replaceString": "    repository: cloudfoundry-community/haproxy-boshrelease\n    version: 11.17.4\n",
            "updates": [
              {
                "bucket": "major",
                "newVersion": "12.1.0+2.7.7",
                "newValue": "12.1.0+2.7.7",
                "releaseTimestamp": "2023-04-28T07:30:47.000Z",
                "newMajor": 12,
                "newMinor": 1,
                "updateType": "major",
                "branchName": "renovate/cloudfoundry-community-haproxy-boshrelease-12.x"
              }
            ],
```

A way for renovate users to diagnose this problem is to try to simulate manually the newVersion replacement, typically within regexp101, as below

https://regex101.com/r/MQDzSp/4

![image](https://github.com/renovatebot/renovate/assets/4748380/7961a763-cd92-453f-b4a6-8ca4218ca4db)

This visual representation highlights that the original currentValue is not matching the simulating newValue, and explains that renovate is then retrying to replace the next extracted depName

Ideally, the error message should state something like `Unable to replace currentValue=11.17.4 with newVersion=12.1.0+2.7.7 check the currentValue rexgexp is properly supporting new versions`

Associated reproduction repo: https://github.com/orange-cloudfoundry/renovate-depName-mismatch-repro

**Comment by HonkingGoose** [comment]
Forked into our `renovate/reproductions` organization: https://github.com/renovate-reproductions/22522
